docs/content/users/extend/custom-compose-files.md

```diff
@@ -76,3 +76,61 @@ When defining additional services for your project, we recommended following the
+
+## Third Party Services May Need To Trust `ddev-webserver`
+
+In some cases, a third-party service (`docker-compose.*.yaml`) may actually consume content from the `ddev-webserver` container. For example, a PDF generator like `gotenberg` might need to consume webserver content to create a PDF, or a testing service may need to do the same.
+
+In this case, the third-party service is by default not configured to trust the certificate authority used by DDEV (via `mkcert`) so you either have to use just HTTP between the two containers, or you have to make the third-party service trust the `mkcert` certificate authority.
+
+The simplest technique is just to use HTTP between the two containers. For example. the [`ddev-selenium-standalone-chrome`](https://github.com/ddev/ddev-selenium-standalone-chrome) service must consume content, and it just conducts interactions with the `ddev-webserver` by accessing `http://web`, see [setup](https://github.com/ddev/ddev-selenium-standalone-chrome/blob/main/config.selenium-standalone-chrome.yaml#L17). In this case, the `selenium-chrome` container just accesses the `web` container via HTTP.
+
+A second technique is to tell the third-party service to ignore HTTPS/TLS errors. For example, if the third-party service uses `curl`, it could use `curl --insecure https://web` or `curl --insecure https://<project>.ddev.site`.
+
+A third and more complex approach is to actually make the third-party container actually trust the self-signed certificate that the `ddev-webserver` is using. This can be done in many cases using a custom Dockerfile and some extra configuration in the `ddev-config.*.yaml`. An example would be:
+
+```yaml
+services:
+  example:
+    container_name: ddev-${DDEV_SITENAME}-example
+    command: "bash -c 'mkcert -install && original-start-command-from-image'"
+    # Add a build stage so we can add mkcert, etc.
+    # The Dockerfile for the build stage goes in the .ddev/example directory here
+    build:
+      context: example
+    environment:
+      - HTTP_EXPOSE=3001:3000
+      - HTTPS_EXPOSE=3000:3000
+      - VIRTUAL_HOST=$DDEV_HOSTNAME
+    # Adding external_links allows connections to `https://example.ddev.site`,
+    # which then can go through ddev-router
+    external_links:
+      - ddev-router:${DDEV_SITENAME}.${DDEV_TLD}
+    labels:
+      com.ddev.approot: $DDEV_APPROOT
+      com.ddev.site-name: ${DDEV_SITENAME}
+    restart: 'no'
+    volumes:
+      - .:/mnt/ddev_config
+      # ddev-global-cache gets mounted so we have the CAROOT
+      # This is required so that the CA is available for mkcert to install
+      - ddev-global-cache:/mnt/ddev-global-cache
+```
+
+```Dockerfile
+FROM example/example
+
+# CAROOT for mkcert to use, has the CA config
+ENV CAROOT=/mnt/ddev-global-cache/mkcert
+
+# If the image build does not run as USER root,
+# temporarily change to root. If the image runs as USER root (default) then
```

Comment:
I’m not confident that I’m reading this as you intend, but it feels like something should need backticks.

> If the image build does not run as USER root

The uppercase “user” here is throwing me off, but I wonder if this should read “Temporarily change to the `root` user in case the image build is run by another.”

Or even simply “Make sure the image build is running as the `root` user.”

Reply:
The upper-case USER is referring to the default 

```Dockerfile
USER root
```

So I don't think the usage here is as good as it could be. 

```suggestion
# If the image build does not run as the default `root` user,
# temporarily change to root. If the image already has the default setup
# where it builds as `root`, then
```